apps/mobile/src/queries/balance/stacks-balance.query.ts

```diff
@@ -0,0 +1,76 @@
+import { useCallback } from 'react';
+
+import { useMempoolTxsBalance } from '@/queries/stacks/use-mempool-txs-balance';
+import { QueryObserverResult, useQueries } from '@tanstack/react-query';
+
+import { Money, StxCryptoAssetBalance } from '@leather.io/models';
+import {
+  AddressBalanceResponse,
+  createGetStacksAccountBalanceQueryOptions,
+  createStxCryptoAssetBalance,
+  createStxMoney,
+  useCryptoCurrencyMarketDataMeanAverage,
+  useCurrentNetworkState,
+  useStacksClient,
+} from '@leather.io/query';
+import { baseCurrencyAmountInQuote, createMoney, sumMoney } from '@leather.io/utils';
+
+interface StxBalances {
+  totalStxBalance: Money;
+  accounts: QueryObserverResult<StxCryptoAssetBalance, unknown>[];
+}
+
+function useGetStxBalanceByAddresses(addresses: string[]): StxBalances {
+  const { totalData, queries } = useStxBalancesQueries(addresses);
+
+  if (!addresses || addresses.length === 0)
+    return { totalStxBalance: createMoney(0, 'STX'), accounts: [] };
```

Comment:
Defaulting to `0` here is wrong. We need to prioritise [this issue](https://linear.app/leather-io/issue/LEA-505/loading-state-for-balances-on-home-screen)